code/datums/components/lightdrawn.dm

```diff
@@ -0,0 +1,132 @@
+
+/**
+  * Drawn to light component
+  *
+  * This component adds game mechanics to emulate the moth chasing a lamp meme
+  */
+/datum/component/lightdrawn
+	/// Next time we can move
+	var/next_move
+	/// How often we move
+	var/move_delay = 2 SECONDS
+	/// Limiter to stop LAMP spam
+	var/next_message
+	/// current light mood
+	var/light_mood = 100
+	/// threshold below which lamp memes happen
+	var/light_mood_threshold = 50
+	/// threshold to compare with [/atom/movable/lighting_object/var/cached_max]
+	var/light_level_threshold = 1
+	/// boolean to track druggy/mesmerised overlay add/remove
+	var/mesmerised = FALSE
+
+/datum/component/lightdrawn/Initialize()
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	. = ..()
+
+	START_PROCESSING(SSprocessing, src)
+
+/datum/component/lightdrawn/Destroy(force, silent)
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/datum/component/lightdrawn/RegisterWithParent()
+	. = ..()
+	RegisterSignal(parent, COMSIG_LIVING_FLASH_ACT, VARSET_CALLBACK(src, light_mood, 100))
+
+/datum/component/lightdrawn/UnregisterFromParent()
+	. = ..()
+	UnregisterSignal(parent, COMSIG_LIVING_FLASH_ACT)
+
+/datum/component/lightdrawn/process()
+	var/turf/T = get_turf(parent)
+	if(!T)
+		return
+
+	var/mob/living/living_parent = parent
+
+	///The amount our ssight is tinted by, such as glasses
+	var/tint_factor = 1
+	///The brightness multplier for the seen brightness
+	var/brightness_mult = 1
+
+	if(iscarbon(parent))
+		var/mob/living/carbon/C = parent
+		var/totaltint = 0
+		totaltint = C.get_total_tint()
+		tint_factor = 1/(1 + totaltint)
+
+	///The actual brightness we "see"
+	var/apparent_brightness = T.lighting_object.cached_max * tint_factor * brightness_mult
+
+	// update their mood
+	if(living_parent.loc != T) // inside something
+		apparent_brightness = 0 // its dark inside things
+		light_mood -= 4
+	else
+		var/change = ((apparent_brightness - light_level_threshold) ** 2)*10
+
+		if(apparent_brightness < light_level_threshold)
+			light_mood -= change
+		else
+			change *= 1.5	//Regen lightmood faster than we lose it so we dont spend 5 minutes running at a lamp
+			light_mood += change
+
+	light_mood = clamp(light_mood, 0, 100)
+	switch(light_mood)
+		if(0 to light_mood_threshold * 0.5)
+			SEND_SIGNAL(parent, COMSIG_ADD_MOOD_EVENT, "missing_the_light", /datum/mood_event/missing_the_light_a_lot)
+		if((light_mood_threshold * 0.5) to light_mood_threshold)
+
+			SEND_SIGNAL(parent, COMSIG_ADD_MOOD_EVENT, "missing_the_light", /datum/mood_event/missing_the_light)
+
+	if(light_mood >= light_mood_threshold)
+		if(mesmerised)
+			living_parent.clear_fullscreen("mesmerised")
+			SEND_SIGNAL(living_parent, COMSIG_CLEAR_MOOD_EVENT, "missing_the_light")
+			mesmerised = FALSE
+		return	// mood is okay, done here
+
+	if(apparent_brightness < light_level_threshold)
+		return  // not bright enough, done here
+
+	living_parent.Stun(1 SECONDS)

```

Comment:
this 👏 still 👏 hardstuns 👏 

and again, what if I wanna play batmoth?

Reply:
All the better,  literally never go into the light.

I dont see the issue with stunning, otherwise you'll just run around and negate the entire desired effect